packages/nimble-components/src/select/specs/IxD.md

```diff
@@ -0,0 +1,286 @@
+| `END`                  | Opens the dropdown and moves visual focus to the last list option                                                                                                                       |
+| _Printable characters_ | Opens the dropdown (if closed), moves visual focus to the filter input (if filterable) with the characters typed or to the first list option that matches the typed character or string |
+
+`TAB` focuses the select component from the previous or next focusable element on the page.
+
+![ ](spec-images/select-26.png)
+
+`ENTER` or `SPACE` on the focused component opens the dropdown and maintains current focus and (if applicable) list option selection.
+
+![ ](spec-images/select-27.png)
+
+`DOWN ARROW` on the focused component opens the dropdown and maintains current focus and (if applicable) list option selection.
+
+![ ](spec-images/select-28.png)
+
+> NOTE: NEEDS DEV REVIEW & VISUAL DESIGN?
+
+`UP ARROW` on the focused component opens the dropdown and focuses (but does not select) the first or last list option.
+
+![ ](spec-images/select-29.png)
+
+#### Dropdown
+
+> QUESTION: WHAT KEYBOARD SHORTCUT SHOULD BE USED TO FOCUS THE FIRST LIST OPTION AFTER THE FILTER BAR?
+
+| Key                        | Description                                                                                                                                                                 |
+| -------------------------- | --------------------------------------------------------------------------------------------------------------------------------------------------------------------------- |
+| `ENTER` or `SPACE`         | Sets the value of the field button to the focused list option, closes the dropdown, and moves focus to the select                                                           |
+| `DOWN ARROW` or `UP ARROW` | Moves visual focus to the next or previous list option                                                                                                                      |
+| `ESC`                      | Closes the dropdown and moves visual focus to the select                                                                                                                    |
+| `HOME` or `END`            | Moves visual focus to the first or last list option                                                                                                                         |
+| `PAGEUP` or `PAGEDOWN`     | Jumps visual focus up or down ten list options (or to first or last list option)                                                                                            |
+| _Printable characters_     | Moves visual focus to the filter input (if filterable) with the characters typed or to the first list option (if not filterable) that matches the typed character or string |
```

Comment:
I feel like if focus moves to the filter input there will be an instinct to press tab to move out of the filter input to the option items. Is that a concern?

Is the interaction clear for typing a few characters, pressing down and up a bit, and then trying to edit the filter again? I think I'd expect after typing characters and having the focus jump to the filter input that pushing up would select the last itme, and pushing down would move focus back to the filter input

Reply:
I can't really picture what you're describing. Can you rephrase this a bit?

Reply:
I've tweaked the proposal to have the fiter input maintaining the text focus to resolve the concerns about focus jumping between the filter bar and the list option. Also, with `Tab`, I'm proposing that it behave like `Esc` – i.e. not a valid way to select an item, but just cancel out of the dialog.